### ISSUE TITLE
A hidden target should be valid in the rule "WCAG20_Label_RefValid"

### ISSUE THREAD

**Issue by aliunwala** [IBMa/equal-access#921]
Hidden elements are not included in accessibility tree, but hey are included in the accessible name or description calculated if they are referred by native label, aria-label, aria-labelledby or aria-describeredby.  The [spec](https://www.w3.org/TR/accname-1.2/) reads:

```
If the current node is hidden and is not directly referenced by aria-labelledby or aria-describedby, nor directly referenced by a native host language text alternative element (e.g. label in HTML) or attribute, return the empty string.
Comment:
By default, assistive technologies do not relay hidden information, but an author can explicitly override that and include hidden text as part of the accessible name or accessible description by using aria-labelledby or aria-describedby.
```
 The help content for the rule "WCAG20_Label_RefValid" indicates the rule is for:

 `The `for` attribute must reference a non-empty, unique `id` attribute of an `<input>` element`

   and it doesn't mention the 'hidden' target at all.

The issue with a hidden target is reported in another rule "WCAG20_Label_TargetInvisible" as a potential violation. 

## This ticket is for the following (select only one): 

- [ ] `New test cases`
- [ ] `Update to a current rule`
- [ ] `Helping text for a rule`
- [ ] `other`

![image](https://user-images.githubusercontent.com/4451128/171962692-4b92b1bc-a7cc-4387-aa8c-16ea12169a80.png)

**Comment by jokerulez** [comment]
I've found this exact issue, it is definitively still a problem

**Comment by MHoov** [comment]
Verify we are using the new name accessible calculation,
